spec/tictactoe_spec.rb

```diff
@@ -1,26 +1,123 @@
 require 'spec_helper' 
 require 'tictactoe' 
+## Faker Classes 
+class FakeIO
+  attr_accessor :stdin, :stdout 
+  def initialize(stdin: StringIO.new('') ) 
+      @stdin = stdin 
+      @stdout = []  
+  end 
+
+  def puts( string ) 
+    @stdout << string     
+  end 
+
+  def gets 
+    @stdin.gets @stdin.string 
+  end 
+end 
+
+def isRowOutputed( io: , row:  ) 
+    expect( io.stdout.include? row).to eql true 
+end 
+
 describe TicTacToe do 
-  context 'when the board is viewed' do
+
+  context 'when the board is rendered' do
     it 'returns a single list with 9 elements' do 
-     game = TicTacToe.new
+     game = TicTacToe.new(io: FakeIO.new)
 
      expect(game.renderBoard.length).to eql 9
-    end 
-
-    it 'returns a list of all strings' do 
-     game = TicTacToe.new
+     expect(game.renderBoard.all?{ |space| space.class == String && space.length == 1 } ).to eql true 
+    end
 
-     expect(game.renderBoard.all?{ |space| space.class == String && space.length <= 1 } ).to eql true 
-    end 
   end
 
   context 'when the game starts the board is empty' do 
-    it 'sets the board to a list with empty strings' do 
-        game = TicTacToe.new
+    it 'sets the board to a list with a single space' do 
+        game = TicTacToe.new(io: FakeIO.new)
         game.newGame 
-        expect( game.renderBoard.all?{ |space| space == '' } ).to eql true
+        expect( game.renderBoard.all?{ |space| space == ' ' } ).to eql true
+    end 
+  end  
+
+ context 'when a single token is placed on an empty board' do 
+    it 'the token will be "X"' do 
+      game = TicTacToe.new(io: FakeIO.new) 
+      expect( game.currentPlayer).to eql 'X' 
+    end
+
+    it 'the specified space will place the current token on the board' do 
+      fakeIO = FakeIO.new 
+      game = TicTacToe.new(io: fakeIO)  
+      game.placeToken(5) 
+      
+      isRowOutputed(io: fakeIO, row:" |X| ") 
+    end
+
+    it 'the token can be placed anywhere on the board' do 
+      fakeIO = FakeIO.new
+      expectedRowOutput = ""
+      game = TicTacToe.new(io: fakeIO)
+
+      game.placeToken(3)
+      isRowOutputed(io: fakeIO, row:" | |X" )  
+      
+      game.newGame
+      fakeIO.stdout = []
```

Comment:
Is it necessary to reset `stdout`? I am wondering some of these can be combined?

```ruby
TicTacToe.new(io: io)
game.place_token(3)
game.place_token(5)
game.place_token(7)

is_row_outputed(io: io, row: " | |X")
is row_outputed(io: io, row: " |X| ")
is_row_outputed(io: io, row: " |X| ")
```

Also, the convention is Ruby is to use [Snake Case](https://en.wikipedia.org/wiki/Snake_case) vs Camel Case

Reply:
Yes. Without it, a row that satisfies the test for a particular position can already be in the `stdout`, yielding a false positive. For instance, if I test position 1 the string `" |X| "` will be added to `stdout`. Then if I test position 4, it also has a row arrangement of `" |X| "`. Because of this, the test for position 4 will pass before it's been written.

This is an issue with FakeIO since it doesn't automatically clear `stdout` like the real IO.

Reply:
I think this is how `stdout` would work, wouldn't? All of the previous output from the game is still on the screen, even if you `clear`, right? (You can scroll back to see the older output.)

Reply:
Oh, I see what you mean. I confused stdout with $stdin.string since that clears the stream after its used.  I'll implement a second check to ensure the space is taken instead of just rendered.